src/main/java/edu/ucsb/ucsbcslas/controllers/TutorNotesController.java

```diff
@@ -0,0 +1,274 @@
+import org.springframework.web.bind.annotation.RestController;
+
+import org.json.JSONObject;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.validation.Valid;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.ucsb.ucsbcslas.advice.AuthControllerAdvice;
+import edu.ucsb.ucsbcslas.models.Course;
+import edu.ucsb.ucsbcslas.entities.Tutor;
+import edu.ucsb.ucsbcslas.entities.TutorNotes;
+import edu.ucsb.ucsbcslas.repositories.CourseRepository;
+import edu.ucsb.ucsbcslas.repositories.TutorRepository;
+import edu.ucsb.ucsbcslas.repositories.TutorNotesRepository;
+
+@RestController
+public class TutorNotesController {
+    private final Logger logger = LoggerFactory.getLogger(TutorNotesController.class);
+
+    @Autowired
+    private AuthControllerAdvice authControllerAdvice;
+    @Autowired
+    private TutorNotesRepository tutorNotesRepository;
+    @Autowired
+    private CourseRepository courseRepository;
+    @Autowired
+    private TutorRepository tutorRepository;
+
+    private ObjectMapper mapper = new ObjectMapper();
+
+    private ResponseEntity<String> getUnauthorizedResponse(String roleRequired) throws JsonProcessingException {
+        Map<String, String> response = new HashMap<String, String>();
+        response.put("error", String.format("Unauthorized; only %s may access this resource.", roleRequired));
+        String body = mapper.writeValueAsString(response);
+        return new ResponseEntity<String>(body, HttpStatus.UNAUTHORIZED);
+    }
+
+    private ResponseEntity<String> getIncorrectInputResponse() throws JsonProcessingException {
+        Map<String, String> response = new HashMap<String, String>();
+        response.put("error", String
+                .format("Misformatted Input; Check that the tutor email that was input is assigned to a valid tutor"));
+        String body = mapper.writeValueAsString(response);
+        return new ResponseEntity<String>(body, HttpStatus.BAD_REQUEST);
+    }
+
+    @PostMapping(value = "/api/member/tutorNotes", produces = "application/json")
+    public ResponseEntity<String> createTutorNotes(@RequestHeader("Authorization") String authorization,
+            @RequestBody @Valid String tutorNotes) throws JsonProcessingException {
+        if (!authControllerAdvice.getIsMember(authorization)) {
+            return new ResponseEntity<>("Unauthorized Request", HttpStatus.UNAUTHORIZED);
+        }
+        if (!authControllerAdvice.getIsAdmin(authorization)) {
+            if (courseRepository.findAllByInstructorEmail(authControllerAdvice.getUser(authorization).getEmail())
+                    .isEmpty()) {
+                return getUnauthorizedResponse("instructor");
+            }
+        }
+
```

Comment:
I'm a little confused by this authentication. It seems that you're intending to only let users who have both the member and admin roles, and are registered as instructors, to view this page. Is this the intended behavior of this endpoint?